x/evmutil/keeper/evm.go

```diff
@@ -118,11 +120,13 @@ func (k Keeper) CallEVMWithData(
 		true,                  // checkNonce
 	)
 
-	res, err := k.evmKeeper.ApplyMessage(ctx, msg, evmtypes.NewNoOpTracer(), true)
+	res, err := k.evmKeeper.ApplyMessage(ethGasContext, msg, evmtypes.NewNoOpTracer(), true)
 	if err != nil {
 		return nil, err
 	}
 
+	ctx.GasMeter().ConsumeGas(res.GasUsed, "evm gas consumed")
```

Comment:
@rhuairahrighairidh @DracoLi possible we should consume gas after the res.Failed() check?